router/contract.ts

```diff
@@ -0,0 +1,36 @@
+import { initContract } from "@ts-rest/core";
+import { Game } from "../types/game";
+import { Result } from "../types/result";
+
+const c = initContract();
+
+export const rpsContract = c.router({
+  start: {
+    method: "GET",
+    path: "/start",
+    responses: {
+      200: c.type<{ instructions: string }>(),
+    },
+    summary: "Get the instructions for rock-paper-scissors",
+  },
+
+  play: {
+    method: "POST",
+    path: "/play",
+    body: c.type<{ move: "string" }>(),
```

Comment:
Tipicamente noi useremmo sempre dei codec zod sia per gli input (come `body`) che per gli output (`responses`) delle rotte.

Questo perché, se usi zod per descrivere i tipi degli input, ts-rest li usa per fare la validazione direttamente, così nell'handler della rotta non hai bisogno di farla e non serve gestire l'errore.

Per esempio, prova a mandare un body non valido perché non ha proprio il campo `move` oppure uno che ha il campo `move` ma contiene un valore non valido. Ora, nel primo caso avrai un 500 perché ts-rest si è "fidato" che il campo `move` ci sia, ma poi il tuo handler lo ha trovato undefined. Nel secondo caso avrai un 400 dato da `Move.safeParse` che chiami nell'handler in `app.ts`.

Poi prova a modificare questa riga in
```suggestion
    body: z.object({ move: z.string() }),
```
oppure
```suggestion
    body: z.object({ move: Move }),
```

Nel primo caso ts-rest, usando quel codec, ti gestirà bene il caso in cui il body non ha proprio il campo `move`. Nel secondo, ti gestirà anche il caso in cui non sia un valore non valido: nell'handler hai direttamente una `Move` e quindi non ti serve più chiamare la `Move.safeParse`.